src/components/VersionSelector/VersionSelector.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import {
+  ArrowIcon,
+  StyledWrapper,
+  StyledButtonWrapper,
+  StyledButton,
+  StyledLabel,
+  StyledDescription,
+  StyledMenuList,
+  StyledDisplay,
+  StyledDropdown,
+} from './styled.elements';
+import { Option } from './Option';
+import { VersionSelectorProps } from './types';
+import { useOutsideClick } from './use-outside-click';
+
+/**
+ * Version Selector Dropdown component based structurally and stylistically off LG Select
+ */
+const VersionSelectorComponent = ({
+  resourceVersions,
+  active,
+  description,
+}: VersionSelectorProps): JSX.Element => {
+  const initialSelectedIdx = resourceVersions.indexOf(active.resourceVersion);
+  const [open, setOpen] = React.useState<boolean>(false);
+  const [selectedIdx, setSelectedIdx] = React.useState<number>(initialSelectedIdx);
+  const menuListRef = React.useRef(null);
+  useOutsideClick(menuListRef, () => {
+    if (open) setOpen(false);
+  });
+
+  const handleClick = (idx: number) => {
+    setSelectedIdx(idx);
+    setOpen(false);
+  };
+
+  return (
+    <StyledWrapper ref={menuListRef}>
```

Comment:
out of curiosity, would it be possible to provide a function handler to the `onClick` property of the `StyledWrapper`?

For example:

```tsx
  <StyledWrapper onClick={()=>setOpen(!open)}>
```

Reply:
We could, however the `StyledWrapper` is actually a bigger div surrounds the `label`, the `description`, the `button`, and the `ul` (when the `ul` is visible). I think for UX, I usually lean towards only having the button itself be clickable - reducing unexpected interactions. 
Also, looking at it again, I feel like I named `StyledButtonWrapper` deceptively (read: _badly_). I'll change that!

Reply:
Ok gotcha, what is the goal of the `menuListRef`?

Reply:
Nope, you're right! It truly is just used to be a reference for that `useOutsideClick` hook. Just making sure that any click outside of the entire `VersionSelector` would close the dropdown section if open. I modularized it, but I suppose it might not be reused unless we go wild with custom components later - but rolling it out is nice on the eyes and more future-proof. Do you think this should we done another way?

Reply:
That makes sense! I didn't fully understand the purpose of it at first, but this clarifies it for me. Thanks!!

Reply:
Of course! Glad you asked :)